__tests__/InvalidMutations.ts

```diff
@@ -43,23 +43,49 @@ class R1 extends Reactor {
         this.addMutation(
             new Triggers(this.in1),
             new Args(this.in1, this.out2),
-            function(this, __in, __out) {
-                // test('expect error to be thrown on mutation creating loop', () => { 
+            function(this, __in1, __out2) {
+                test('expect error to be thrown on mutation creating loop', () => { 
+                    expect(() => {
+                        this.connect(__out2, __in1)
+                    }).toThrowError("New connection introduces cycle.")
+                    expect(() => {
+                        this.connect(__in1, __out2)
+                    }).toThrowError("New connection introduces direct feed through.") // dist port already in use
+                    
+                    // let R2 = new R1(this.getReactor())
+                    // expect(() => {
+                    //     // this.connect(R2.out1, R2.in1)
+                    //     // this.connect(R2.in1, R2.out1)
+                    //     // this.connect(R2.in1, R2.out1)
+                    //     this.connect(R2.out1, R2.in1)
+                    // }).toThrowError("New connection introduces cycle.")
+                })
+                // console.log("TEST START!")
+                // test('expect error on mutation creating race condition', () => {
                 //     expect(() => {
-                //         this.connect(__out, __in)
-                //     }).toThrowError("ERROR connecting " + __out + " to " + __in)
+                //         //this.canConnect(__in1)
+                //         // console.log("Can Connect: ", this.getReactor().canConnect(__in1, __out2))
+                        
+                //         this.connect(__in1, __out2)
+                //     }).toThrowError("New connection introduces direct feed through.")
+                // })
+                // let R2 = new R1(this.getReactor())
+                // test('expect error on spawning and creating loop within a reactor', () => {
+                //     expect(() => {
+                //         this.connect(__out2, __in1)
+                //     }).toThrowError("New connection introduces cycle.")
```

Comment:
The commented lines should be cleaned up.

Reply:
It was reflected in https://github.com/lf-lang/reactor-ts/pull/108/commits/9019da3204bf0e1773451d2c1a27b0485e3ac8e1